### PR TITLE
38 extend expressionatlasfetcher for summary function

### PR DESCRIPTION
**Pull request by alexgrent**
used summary function from abstract dataset fetcher, returns list of dictionary with keys:        

"id": experiment['experimentAccession'],
"title": experiment['experimentDescription'],
"species": experiment['species'],
"no_samples": experiment['numberOfAssays'],
"technology": experiment['technologyType']

### CODE REVIEW

File: reactome_analysis_datasets/reactome_analysis_datasets/dataset_fetchers/expression_atlas_fetcher.py [hunk 1]
```diff
@@ -30,6 +31,9 @@
 import os
 import time
 import signal
+import requests
+
+from reactome_analysis_api.models.external_data_sample_metadata import ExternalDataSampleMetadata
```
Comment (jgriss) [inline reactome_analysis_datasets/reactome_analysis_datasets/dataset_fetchers/expression_atlas_fetcher.py:36]: Ideally, imports should be in some kind of order (just for the future)
ie. put all internal imports at the top and system imports afterwards or vice versa

Comment (jgriss) [reply reactome_analysis_datasets/reactome_analysis_datasets/dataset_fetchers/expression_atlas_fetcher.py:36]: Btw. I don't think that we need this import at all.

File: reactome_analysis_datasets/reactome_analysis_datasets/dataset_fetchers/expression_atlas_fetcher.py [hunk 2]
```diff
@@ -83,6 +88,50 @@ def get_dataset_id(self, parameters: list) -> str:
...
+        :param no_datasets: number of datasets to retrieve
+        :returns: datasets in ExternalData format
+        """
+        experiments_external_data_list = list()
+        try:
+            experiments_url = "https://www.ebi.ac.uk/gxa/json/experiments"
+            response = requests.get(experiments_url)
+            response.raise_for_status()
+            json_response = response.json()
+            experiments_list = json_response['experiments'][0:no_datasets]
```
Comment (jgriss) [inline reactome_analysis_datasets/reactome_analysis_datasets/dataset_fetchers/expression_atlas_fetcher.py:103]: Here I made a mistake. Similar to the grein_loader function the no_datasets should be changed to

```
def get_available_datasets(self, no_datasets: int=None): -> list
```

Can you adapt the AbstractDatasetFetcher as well?

You'd then have to adapt this line for cases where the no_datasets is not set.

**Review by jgriss — CHANGES_REQUESTED** [review]
Once this is done, we should discuss how to implement this in the API

File: reactome_analysis_datasets/reactome_analysis_datasets/dataset_fetchers/expression_atlas_fetcher.py [hunk 3]
```diff
@@ -83,6 +88,50 @@ def get_dataset_id(self, parameters: list) -> str:
...
+                    "no_samples": experiment['numberOfAssays'],
+                    "technology": experiment['technologyType']
+                }
+                experiments_external_data_list.append(experiment_data_dict)
+        except requests.exceptions.RequestException:
+            LOGGER.error("Response not available")
+
+        return experiments_external_data_list
+
+    def get_dataset_by_id(self, dataset_id):
```
[the remark this answers is not in the document]
Comment (alexgrent) [reply reactome_analysis_datasets/reactome_analysis_datasets/dataset_fetchers/expression_atlas_fetcher.py:119]: No removed the method, I forgot to delete it. The Import from above is therefore not needed.

File: reactome_analysis_datasets/reactome_analysis_datasets/dataset_fetchers/expression_atlas_fetcher.py (hunk 1 above)
Comment (alexgrent) [reply reactome_analysis_datasets/reactome_analysis_datasets/dataset_fetchers/expression_atlas_fetcher.py:36]: True not needed anymore

File: reactome_analysis_datasets/reactome_analysis_datasets/dataset_fetchers/expression_atlas_fetcher.py (hunk 2 above)
Comment (alexgrent) [reply reactome_analysis_datasets/reactome_analysis_datasets/dataset_fetchers/expression_atlas_fetcher.py:103]: changed to a default value of 10000 datasets

**Review by jgriss — APPROVED** [review]

**Pull request merged**